File: lab2pbudhath/app/src/main/java/com/example/lab2_pbudhath/ServiceSelectActivity.java
package com.example.lab2_pbudhath;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class ServiceSelectActivity extends AppCompatActivity {

    public final static String PARAM1 = "com.example.lab2_pbudhath.P1";
    public final static String PARAM2 = "com.example.lab2_pbudhath.P2";
    public final static String PARAM3 = "com.example.lab2_pbudhath.P3";

    private double exc_quality = -1, avg_quality = -1, bavg_quality = -1;
    private double o_exc_quality = 0, o_avg_quality = 0, o_bavg_quality = 0;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_service_select);
        // Get parameters
        Intent intent = getIntent();
        o_exc_quality = intent.getDoubleExtra(this.PARAM1, 0.25);
        o_avg_quality = intent.getDoubleExtra(this.PARAM2, 0.20);
        o_bavg_quality = intent.getDoubleExtra(this.PARAM3, 0.15);
        EditText Exc_Edit = findViewById(R.id.Exc_Edit);
        EditText Avg_Edit = findViewById(R.id.Avg_Edit);
        EditText Bavg_Edit = findViewById(R.id.Bavg_Edit);

        Exc_Edit.setHint(Double.toString(o_exc_quality*100));
        Avg_Edit.setHint(Double.toString(o_avg_quality*100));
        Bavg_Edit.setHint(Double.toString(o_bavg_quality*100));

        TextWatcher tw = new TextWatcher() {
            public void afterTextChanged(Editable s) {
                if (s == Exc_Edit.getEditableText()) {
                    if (Exc_Edit.getText().length() != 0) {
                        exc_quality = Float.parseFloat(String.valueOf(Exc_Edit.getText()))/100;

                        if(exc_quality < 0){
                            Toast.makeText(getApplicationContext(), "Percentage must be a positive integer", Toast.LENGTH_SHORT).show();
                        }
                    }
                }
                else if (s == Avg_Edit.getEditableText()) {
                    if (Avg_Edit.getText().length() != 0) {
                        avg_quality = Float.parseFloat(String.valueOf(Avg_Edit.getText()))/100;

                        if(avg_quality < 0){
                            Toast.makeText(getApplicationContext(), "Percentage must be a positive integer",Toast.LENGTH_SHORT).show();
                        }
                    }
                }
                else if (s == Bavg_Edit.getEditableText()) {
                    if (Bavg_Edit.getText().length() != 0) {
                        bavg_quality = Float.parseFloat(String.valueOf(Bavg_Edit.getText()))/100;

                        if(bavg_quality < 0){
                            Toast.makeText(getApplicationContext(), "Percentage must be a positive integer",Toast.LENGTH_SHORT).show();
                        }
                    }
                }
            }

            public void beforeTextChanged (CharSequence s,int start, int count, int after){
            }
            public void onTextChanged (CharSequence s,int start, int before, int count){
            }

        };
        Exc_Edit.addTextChangedListener(tw);
        Avg_Edit.addTextChangedListener(tw);
        Bavg_Edit.addTextChangedListener(tw);

    }

    public void onClick (View v){
        Intent intent = new Intent();
        if (v.getId() == R.id.Keep_Button){
            if(exc_quality == -1){
                intent.putExtra(MainActivity.RESULT1, o_exc_quality);
            }
            else {
                intent.putExtra(MainActivity.RESULT1, exc_quality);
            }
            if(avg_quality == -1){
                intent.putExtra(MainActivity.RESULT2, o_avg_quality);
            }
            else {
                intent.putExtra(MainActivity.RESULT2, avg_quality);
            }
            if(bavg_quality == -1){
                intent.putExtra(MainActivity.RESULT3, o_bavg_quality);
            }
            else {
                intent.putExtra(MainActivity.RESULT3, bavg_quality);
            }

        }
        if (v.getId() == R.id.Cancel_Button){
            intent.putExtra(MainActivity.RESULT1, o_exc_quality);
            intent.putExtra(MainActivity.RESULT2, o_avg_quality);
            intent.putExtra(MainActivity.RESULT3, o_bavg_quality);
        }
        setResult(Activity.RESULT_OK, intent);
        finish();
    }
}

File: lab7pbudhath/app/src/main/java/com/example/lab7_pbudhath/ServiceSelectActivity.kt
package com.example.lab7_pbudhath

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.EditText
import android.widget.Toast

class ServiceSelectActivity : AppCompatActivity() {

    companion object{
        val PARAM1: String = "PARAM1"
        val PARAM2: String = "PARAM2"
        val PARAM3: String = "PARAM3"
    }

    private var exc_quality: Float = -1f
    private var avg_quality: Float = -1f
    private var bavg_quality: Float = -1f

    private var o_exc_quality: Float = 0f
    private var o_avg_quality: Float = 0f
    private var o_bavg_quality: Float = 0f

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_service_select)

        o_exc_quality = intent.getFloatExtra(PARAM1, 0.25f)
        o_avg_quality = intent.getFloatExtra(PARAM2, 0.20f)
        o_bavg_quality = intent.getFloatExtra(PARAM3, 0.15f)
        val exc_edit = findViewById<EditText>(R.id.Exc_Edit)
        val avg_edit = findViewById<EditText>(R.id.Avg_Edit)
        val bavg_edit = findViewById<EditText>(R.id.Bavg_Edit)


        exc_edit.hint = ((o_exc_quality*100).toString())
        avg_edit.hint = ((o_avg_quality*100).toString())
        bavg_edit.hint = ((o_bavg_quality*100).toString())

        val tw = object : TextWatcher {
            override fun afterTextChanged(s: Editable) {
                if (s == exc_edit.text) {
                    if (exc_edit.text.isNotEmpty()) {
                        exc_quality = exc_edit.text.toString().toFloat()/100

                        if(exc_quality < 0){
                            Toast.makeText(applicationContext, "Percentage must be a positive integer", Toast.LENGTH_SHORT).show()
                        }
                    }
                }
                else if (s == avg_edit.text) {
                    if (avg_edit.text.isNotEmpty()) {
                        avg_quality = avg_edit.text.toString().toFloat()/100

                        if(avg_quality < 0){
                            Toast.makeText(applicationContext, "Percentage must be a positive integer", Toast.LENGTH_SHORT).show()
                        }
                    }
                }
                else if (s == bavg_edit.text) {
                    if (bavg_edit.text.isNotEmpty()) {
                        bavg_quality = bavg_edit.text.toString().toFloat()/100

                        if(bavg_quality < 0){
                            Toast.makeText(applicationContext, "Percentage must be a positive integer", Toast.LENGTH_SHORT).show()
                        }
                    }
                }
            }

            override fun beforeTextChanged (s: CharSequence, start: Int, count: Int, after: Int){
            }
            override fun onTextChanged (s: CharSequence, start: Int, before: Int, count: Int){
            }

        }
        exc_edit.addTextChangedListener(tw)
        avg_edit.addTextChangedListener(tw)
        bavg_edit.addTextChangedListener(tw)
    }

    fun onClick(v: View){
        val intent: Intent = Intent()
        if (v.getId() == R.id.Keep_Button){
            if(exc_quality == -1f){
                intent.putExtra(MainActivity.RESULT1, o_exc_quality);
            }
            else {
                intent.putExtra(MainActivity.RESULT1, exc_quality);
            }
            if(avg_quality == -1f){
                intent.putExtra(MainActivity.RESULT2, o_avg_quality);
            }
            else {
                intent.putExtra(MainActivity.RESULT2, avg_quality);
            }
            if(bavg_quality == -1f){
                intent.putExtra(MainActivity.RESULT3, o_bavg_quality);
            }
            else {
                intent.putExtra(MainActivity.RESULT3, bavg_quality);
            }

        }
        if (v.getId() == R.id.Cancel_Button){
            intent.putExtra(MainActivity.RESULT1, o_exc_quality);
            intent.putExtra(MainActivity.RESULT2, o_avg_quality);
            intent.putExtra(MainActivity.RESULT3, o_bavg_quality);
        }
        setResult(Activity.RESULT_OK, intent);
        finish();
    }
}